plone/app/contenttypes/browser/collection.py

```diff
@@ -13,8 +13,16 @@ class CollectionView(FolderView):
     def __init__(self, *args, **kwargs):
         super(CollectionView, self).__init__(*args, **kwargs)
         context = aq_inner(self.context)
-        self.collection_behavior = ICollection(context)
-        self.b_size = self.collection_behavior.item_count
+
+        try:
+            # the object may have the behavior applied
+            self.behavior = ICollection(context)
```

Comment:
I would not change the name from `self.collection_behavior` to `self.behavior`. The standard templates do not use this, but someone may have customised them and use this name. Changing it would break those templates.